### DISCUSSION TITLE
Inspecting elements for enumeration and custom html elements

### DISCUSSION THREAD

**Discussion by mwpowellhtx** [cefsharp/CefSharp#4868] [Q&A]
Hello,

I consider myself fairly well versed in HTML, elements, fairly proficient at the attributes, styles, and selectors thereby... However, I found what looks to be a custom one, wondered if anyone happened to know what this was...

```html
<a class="..." href="..." aria-expanded="true">
    <l>
        <img width="32" height="32" src="...">
    </l>
    <span class="nav-text">{some text labeling the image}</span>
</a>
```

I know what `<a/>`, `<img/>`, and `<span/>` are.

But what is `<l/>`?

I am aiming for scraping those bits out for this application, and in its context, for what it is, I think I can probably safely ignore it.

I've heard of the possibility with some versions of HTML of custom elements, but I am not familiar with this one. Was just curious.

Best. 🍻

**Comment by mwpowellhtx** [comment]
Q: along similar lines, I have a series of `<li/>` which I may glean from, for example:

```csharp
var liCandidates = (await ulMenu.QuerySelectorAllAsync<HtmlListItemElement>(_S_.Li)).ToArray();
```

But I need to also discern "menu items" from "menu item actions". Both are `<li/>`, and potentially would return as an array of such elements here.

What I need to know is, does query return a depth first search? Or a breadth first search? That would make a hugh difference, potentially, in whether I could simply parse through the array in sequence, or whether I need to do a bit of LINQ filtering on the `HtmlListItemElement` instances first.

**Reply by amaitland** [reply]
QuerySelectorAllAsync effectively calls https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll in the browser itself.

**Comment by amaitland** [comment]
I don't have time myself to answer generic HTML/JavaScript/CSS questions. I'd suggest asking on  stack overflow or similar.

**Reply by mwpowellhtx** [reply]
Fully aware this was the territory in which I was enmeshed. Fine line there, I fully admit. At the moment, I think I have a sane comprehension of the subject DOM in question, and some approaches that seem to be working, or at at least more resistant, if resilient, to pending anticipated elements.
